src/jqLite.js

```diff
@@ -288,6 +288,8 @@ function JQLite(element) {
 
   if (argIsString) {
     jqLiteAddNodes(this, jqLiteParseHTML(element));
+  } else if (isFunction(element)) {
```

Comment:
I didn't want to delay the most common path and jQuery has this check below the selector code. Maybe with JITs it's not a problem.